features/270-next-skycli/api.md

```diff
@@ -113,27 +113,137 @@ access skycli instead of using user account.
 
 ## Config files structure
 
-### ~/.skycli/skyclirc
+### ${XDG_CONFIG_HOME:-$HOME/.config}/skycli/config
 
 - cluster controller endpoint
 - user credentials 
 
-### ./.skyclirc
-- current app
+### ./skygear.yaml
 
-## Discussion
+- config version (If we have new config version in future, skycli will read the
+  version and convert the previous config to the current version.)
+- current app
+- cloud functions configuration
 
-- Current `skycli app view-config` and `skycli app update-config` give the
-greatest flexible to user to update the config. But user need to understand the
-tenant config structure. I think this case is similar to v1 cms config file.
-- Current tenant config is not a simple key value mapping, so we didn't have a
-single command to update particular value. User need to export the current config
-by `skycli app view-config`, update, and upload by `skycli app update-config`.
-- `skycli app view-config` will mask the auth db url, it is also not editable by
-`skycli app update-config`.
-- I think we should not show the secret value in the list comment. But user may
-want to obtain the db credentials after enabling service?
+#### Example
+```
+config_version: 1
+app: myapp
+cf:
+  function1:
+    type: http-handler
+    path: /hello-world
+    env: node
+    src: js/hellow-world
+    secrets:
+      - DATABASE_URL
+    permission:
+      - name: key_required
+      - name: user_required
+static:
+  ...
+```
 
-## Questions
+## Use cases
 
-- How to support email template?
+- User creates app and uses skycli for app configuration without CF
+
+    ```sh
+    # Create app
+    $ skycli app create
+
+    ? What is your app name? myapp
+    Creating app...
+    Your API endpoint: https://myapp.api.skygear.io
+    Your Client API Key: DSOJSDOSDKOJASNLSLC
+    Your Master API Key: FJOADJOFAJOFJOASDJK
+    Created app successfully!
+
+    To setup app project directory, run:
+        skycli app scaffold
+
+    # User can choose whether to setup the project folder now.
+    ? Do you want to setup the project folder now or you can do it later by
+    `skycli app scaffold` command? (Y/n) n
+
+    # Add user to app
+    $ skycli app add-user dev@example.com --app=myapp
```

Comment:
May I know the expected behavior/purpose of `--app=myapp`? If a developer has multiple apps on the cluster, and each app has different CFs, Will the setting `--app=myapp_2` update local skygear.yml dynamically to the config on the cluster? Or just replace the `app: myapp_2` in the skygear.yml.

If it is just a simple text replace in skygear.yml, may I know the use cases?

Thanks

Reply:
I guess @carmenlau meant to say it just temporarily (within the running session) replace the configuration at skygear.yml.

Reply:
yes, it is temporarily within the running session. The use case is for fast switching between different apps. E.g. if the admin need add user to 2 apps. He can simply use `skycli app add-user dev@example.com --app=myapp` and `skycli app add-user dev@example.com --app=myapp2`. And no need to update the `skygear.yaml`

Reply:
Thanks. :)